DarkScript3/RoundTripTool.cs

```diff
@@ -0,0 +1,201 @@
+                        return "";
+                    }
+                }
+                catch (Exception e)
+                {
+                    sb.AppendLine($"/* ------------------- {name} ------------------- */");
+                    sb.AppendLine($"/* {e} */");
+                    Console.WriteLine(name + ": " + e + "\n");
+                    return null;
+                }
+            }
+            foreach (string emevdPath in emevdPaths)
+            {
+                string name = Path.GetFileNameWithoutExtension(Path.GetFileNameWithoutExtension(emevdPath));
+                if (game == "ds3" && name.StartsWith("m2")) continue;
+                Console.WriteLine("--------------------------" + name);
+                EventScripter scripter = new EventScripter(emevdPath, docs);
+                string reg1 = recordText("reg1", name, () => scripter.Unpack());
+                if (reg1 == null) continue;
+                if (args.Contains("reg"))
+                {
+                    if (recordText("reg2", name + "-compile", () => scripter.Pack(reg1, name)) != null)
+                    {
+                        recordText("reg2", name, () => scripter.Unpack());
+                    }
+                }
+                if (args.Contains("fancy"))
+                {
+                    EventCFG.CFGOptions options = args.Contains("min") ? EventCFG.CFGOptions.MIN : EventCFG.CFGOptions.DEFAULT;
+                    scripter = new EventScripter(emevdPath, docs);
+                    FancyEventScripter fes = new FancyEventScripter(scripter, docs, options);
+                    if (args.Contains("unit"))
+                    {
+                        fes.Pack(TEST_CASES);
+                        Console.WriteLine(scripter.Unpack());
+                        Console.WriteLine(fes.Repack(TEST_CASES));
+                        return;
+                    }
+                    string fancy1 = recordText("fancy1", name, () => fes.Unpack());
+                    if (fancy1 != null)
+                    {
+                        if (args.Contains("repack"))
+                        {
+                            recordText("fancy2", name, () => fes.Repack(fancy1));
+                        }
+                        else
+                        {
+                            if (recordText("reg3", name + "-compile", () => fes.Pack(fancy1, name)) != null)
+                            {
+                                recordText("reg3", name, () => scripter.Unpack());
+                            }
+                        }
+                    }
+                }
+            }
+            foreach (KeyValuePair<string, StringBuilder> entry in contents)
+            {
+                using (TextWriter writer = File.CreateText($@"{outDir}\{game}_{entry.Key}.js"))
+                {
+                    writer.Write(entry.Value);
+                }
+            }
+        }
+
+        public static readonly string TEST_CASES = @"
```

Comment:
To keep things clean, please put test cases in a separate file (e.g. `test.js`) and read them on initialization.